sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TableUtils.java

```diff
@@ -612,16 +613,68 @@ public static String[] getKeysFromToken(String token) {
         return keys;
     }
 
-    public static <T> T callWithOptionalTimeout(Supplier<T> callable, ExecutorService threadPool, Duration timeout,
-        ClientLogger logger) {
+    public static Context skip409Logging(Context context) {
+        return context.addData("skip409logging", true);
+    }
+
+    public static <T> Response<T> callWithOptionalTimeout(Supplier<Response<T>> callable, ExecutorService threadPool, Duration timeout, ClientLogger logger) {
+        try {
+            return callHandler(callable, threadPool, timeout, logger);
+        } catch (Throwable thrown) {
+            Throwable exception = mapThrowableToTableServiceException(thrown);
+            throw logger.logExceptionAsError((RuntimeException) exception);
+        }
+
+    }
+
+    public static <T> Response<T> callWithOptionalTimeoutAndContext(Supplier<Response<T>> callable, ExecutorService threadPool, Duration timeout, ClientLogger logger, Context context) {
+        try {
+            return callHandler(callable, threadPool, timeout, logger);
+        } catch (Throwable thrown) {
+            Throwable exception = mapThrowableToTableServiceException(thrown);
+            context = setContext(context);
+
+            if (exception instanceof TableServiceException) {
+                if (shouldSkip409Logging((TableServiceException) exception, context)) {
+                    // return empty response
+                    HttpResponse resp = ((TableServiceException) exception).getResponse();
+                    return new SimpleResponse<>(resp.getRequest(), resp.getStatusCode(), resp.getHeaders(), null);
+                }
+            }
+
+            throw logger.logExceptionAsError((RuntimeException) exception);
+        }
+    }
+
+    public static <T> PagedIterable<T> callIterableWithOptionalTimeout(Supplier<PagedIterable<T>> callable, ExecutorService threadPool, Duration timeout, ClientLogger logger) {
```

Comment:
I think we can re-merge all the utility methods back down to `callHandler` and just pass the appropriate flag to ignore logging a 409 exception

Reply:
@alzimmermsft I thought about doing that but here's where I'm mentally at on that:

Since the `callHandler()` method is not `void`, the method needs to either need to return `Response<T>` or throw an exception. Previously when the `callWithOptionalTimeout` handled all scenarios the return type was `T`. The current functionality of `TableServiceClient.createTableIfNotExistsWithResponse()` is that when the Table already exists it will return the service's 409 response with an empty response value. This is the expected behavior and cannot change.

Along with not logging, the handler is responsible for returning the service's response. Previously, the exception was being rethrown so it could be caught and handled by the method `createTableIfNotExistsWithResponse()` but this design is what led to the logging issue. So my options in terms of design we're to:

1. Break up the handler into multiple parts allowing for a return type of `Response<T>` to be assigned to the one of the returning methods to allow the handling of the exception (without logging) and returning of the proper response.
2. Combine it down into a single handler that was then made to throw an exception so that the response could be handled properly in the necessary use cases.

Depending on how much of the handler methods get merged down, option 2 would result in a lot of try-catch statements, and so I chose option 1 for that reason.